src/partition/mod.rs

```diff
@@ -43,14 +50,28 @@ impl<'a> SumFilteredPartitionIterator<'a> {
         }
     }
 
+    /// Complexity is $O(\mathrm{FML})$, also in the average case.
```

Comment:
What do you mean by $O(\mathrm{FML})$? "Wowo this is so high or should just end things?" Filter, Memory, Length?

Reply:
I'd understand if it were in plain language saying ~"Exponential, hard to calculate exactly, doesn't really matter"